kubernetes/data_source_kubernetes_mutating_webhook_configuration.go

```diff
@@ -0,0 +1,115 @@
+				Description: apiDoc["webhooks"],
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"admission_review_versions": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["admissionReviewVersions"],
+							Computed:    true,
+							Optional:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						"client_config": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["clientConfig"],
+							Computed:    true,
+							Elem: &schema.Resource{
+								Schema: webhookClientConfigFields(),
+							},
+						},
+						"failure_policy": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["failurePolicy"],
+							Computed:    true,
+						},
+						"match_policy": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["matchPolicy"],
+							Computed:    true,
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["name"],
+							Required:    true,
+						},
+						"namespace_selector": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["namespaceSelector"],
+							Computed:    true,
+							Optional:    true,
+							Elem: &schema.Resource{
+								Schema: labelSelectorFields(true),
+							},
+						},
+						"object_selector": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["objectSelector"],
+							Computed:    true,
+							Optional:    true,
+							Elem: &schema.Resource{
+								Schema: labelSelectorFields(true),
+							},
+						},
+						"reinvocation_policy": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["reinvocationPolicy"],
+							Computed:    true,
+						},
+						"rule": {
+							Type:        schema.TypeList,
+							Description: webhookDoc["rules"],
+							Computed:    true,
+							Elem: &schema.Resource{
+								Schema: ruleWithOperationsFields(),
+							},
+						},
+						"side_effects": {
+							Type:        schema.TypeString,
+							Description: webhookDoc["sideEffects"],
+							Computed:    true,
+							Optional:    true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(admissionregistrationv1.SideEffectClassUnknown),
+								string(admissionregistrationv1.SideEffectClassNone),
+								string(admissionregistrationv1.SideEffectClassSome),
+								string(admissionregistrationv1.SideEffectClassNoneOnDryRun),
+							}, false),
+						},
+						"timeout_seconds": {
+							Type:        schema.TypeInt,
+							Description: webhookDoc["timeoutSeconds"],
+							Computed:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceKubernetesMutatingWebhookConfigurationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("metadata.0.name").(string)
+	d.SetId(name)
+
+	return resourceKubernetesMutatingWebhookConfigurationRead(ctx, d, meta)
```

Comment:
Just realized this should be `resourceKubernetesMutatingWebhookConfigurationV1Read`.

Reply:
@jrhouston  It doesn't appear that any of the other data sources access the `v1Read` versions of their companion resource Read functions. There's also only 1 v1 data source file (ingress); are you good with the mutating webhookconfiguration being different, or are there other changes you'd like me to make?

Reply:
Yeah the rest are basically just aliases for the moment since they were already at v1 – MutatingWebhookConfiguration does have a read function that is v1 specific though so we can use it in this data source. It will still work as is now, just that it would have the undocumented behaviour of also supporting v1beta1.

Reply:
Sounds good; just knocked it out